internal/auth/ldap/repository_auth_method_read.go

```diff
@@ -206,6 +225,44 @@ func (r *Repository) getAuthMethods(ctx context.Context, authMethodId string, sc
 	return authMethods, nil
 }
 
+// ListDeletedAuthMethodIds lists the public IDs of any auth methods deleted since the timestamp provided.
+func (r *Repository) ListDeletedAuthMethodIds(ctx context.Context, since time.Time, options ...auth.Option) ([]string, error) {
+	const op = "ldap.(Repository).ListDeletedAuthMethodIds"
+	opts, err := auth.GetOpts(options...)
+	if err != nil {
+		return nil, err
+	}
+	rd := r.reader
+	if opts.WithReader != nil {
+		rd = opts.WithReader
+	}
+	var deletedAuthMethods []*deletedAuthMethod
+	if err := rd.SearchWhere(ctx, &deletedAuthMethods, "delete_time >= ?", []any{since}); err != nil {
+		return nil, errors.Wrap(ctx, err, op, errors.WithMsg("failed to query deleted auth methods"))
+	}
+	var authMethodIds []string
+	for _, am := range deletedAuthMethods {
+		authMethodIds = append(authMethodIds, am.PublicId)
+	}
+	return authMethodIds, nil
+}
+
+// EstimatedAuthMethodCount returns and estimate of the total number of items in the auth methods table.
+func (r *Repository) EstimatedAuthMethodCount(ctx context.Context) (int, error) {
+	const op = "ldap.(Repository).EstimatedAuthMethodCount"
+	rows, err := r.reader.Query(ctx, estimateCountLdapAuthMethods, nil)
+	if err != nil {
+		return 0, errors.Wrap(ctx, err, op, errors.WithMsg("failed to query total auth methods"))
+	}
+	var count int
+	for rows.Next() {
+		if err := r.reader.ScanRows(ctx, rows, &count); err != nil {
+			return 0, errors.Wrap(ctx, err, op, errors.WithMsg("failed to query total auth methods"))
+		}
+	}
+	return count, nil
+}
+
```

Comment:
Note to @mikemountain; we need to actually implement the logic for the new `StartPageAfterItem` in `getAuthMethod` below.